applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -504,6 +506,33 @@ static void handle_dfu_bootloader_variant(uint8_t *data, size_t *size)
 	strcpy((char *)data, BOOTLOADER_NAME);
 }
 
+static void handle_dfu_devinfo(uint8_t *data, size_t *size)
+{
+	LOG_INF("Device information requested");
+	const uint16_t vid = CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_VID;
+	const uint16_t pid = CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_PID;
+	const char *generation = CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_GENERATION;
+	size_t pos = 0;
+
+	BUILD_ASSERT(sizeof(CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_GENERATION) > 1,
+		     "CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_GENERATION cannot be an empty string");
+	BUILD_ASSERT((sizeof(vid) + sizeof(pid) +
+		      sizeof(CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_GENERATION)) <=
+		     CONFIG_CHANNEL_FETCHED_DATA_MAX_SIZE,
+		     "CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_GENERATION is too long");
+
+	sys_put_le16(vid, &data[pos]);
+	pos += sizeof(vid);
+
+	sys_put_le16(pid, &data[pos]);
+	pos += sizeof(pid);
+
+	strcpy(&data[pos], generation);
```

Comment:
You can consider using the TLV (Type, Length, Value) approach for each parameter. It is easier to maintain backward compatibility this way, as you may add a new type to the devinfo response packet that older versions of response packet parsers will ignore.

The lighter version would be to add a length field for the generation parameter as it will be possible to add another parameter after the Generation field. Currently, the generation value is considered to be the remaining bytes of the packet message and there is no way of adding new field in the future without breaking the compatibility.

Reply:
The configuration channel response can contain only up to the 16 bytes (in our case it's 4 bytes for VID+PID and remaining bytes can be used for the generation string). Using TLV could be more generic approach but it would introduce an additional overhead. Also we already have some options that return multiple values one after another in predefined format (e.g. `fwinfo`). The values returned for given option should not change, if needed we can always introduce a new configuration channel module's option to handle new data.

Please also keep in mind that our Python scripts rely on the fact that incoming data has predefined length in some cases to they may not be interoperable with the proposed expandable message format.

Reply:
Right, 16 bytes seem too low for this type of format. Thanks for the detailed clarification